From 0a5f5b6409080c925a84f60fc9e2de8d40a0f84b Mon Sep 17 00:00:00 2001
From: Vineet Gupta <vgupta@synopsys.com>
Date: Mon, 30 May 2016 15:19:22 +0530
Subject: [PATCH 5/8] xxx don't convert _DYNAMIC@ to GOT

Signed-off-by: Vineet Gupta <vgupta@synopsys.com>
---
 gas/config/tc-arc.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gas/config/tc-arc.c b/gas/config/tc-arc.c
index 8f7da08c2a11..6ac266ea3ffa 100644
--- a/gas/config/tc-arc.c
+++ b/gas/config/tc-arc.c
@@ -3301,10 +3301,10 @@ md_undefined_symbol (char *name)
      GOTPC reference to _GLOBAL_OFFSET_TABLE_.  */
   if (((*name == '_')
        && (*(name+1) == 'G')
-       && (strcmp (name, GLOBAL_OFFSET_TABLE_NAME) == 0))
-      || ((*name == '_')
+       && (strcmp (name, GLOBAL_OFFSET_TABLE_NAME) == 0)))
+/*      || ((*name == '_')
 	  && (*(name+1) == 'D')
-	  && (strcmp (name, DYNAMIC_STRUCT_NAME) == 0)))
+	  && (strcmp (name, DYNAMIC_STRUCT_NAME) == 0)))*/
     {
       if (!GOT_symbol)
 	{
-- 
2.7.4

