From 8a004ff8b35dcc32996b006be9f55f3e9e3f8226 Mon Sep 17 00:00:00 2001
From: Evgeniy Didin <Evgeniy.Didin@synopsys.com>
Date: Tue, 7 Apr 2020 00:16:25 +0300
Subject: [PATCH] python2/libffi: add ARC support

Signed-off-by: Evgeniy Didin <Evgeniy.Didin@synopsys.com>
---
 Modules/_ctypes/libffi/fficonfig.py.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Modules/_ctypes/libffi/fficonfig.py.in b/Modules/_ctypes/libffi/fficonfig.py.in
index d102498..eee0ab3 100644
--- a/Modules/_ctypes/libffi/fficonfig.py.in
+++ b/Modules/_ctypes/libffi/fficonfig.py.in
@@ -28,6 +28,7 @@ ffi_platforms = {
     'PA': ['src/pa/linux.S', 'src/pa/ffi.c'],
     'PA_LINUX': ['src/pa/linux.S', 'src/pa/ffi.c'],
     'PA_HPUX': ['src/pa/hpux32.S', 'src/pa/ffi.c'],
+    'ARC': ['src/arc/sysv.S', 'src/arc/ffi.c', 'src/arc/arcompact.s'],
 }
 
 ffi_sources += ffi_platforms['@TARGET@']
-- 
2.16.2

