From 751188d79ed6367f8717672dd0637eedecee2f9c Mon Sep 17 00:00:00 2001
From: Cupertino Miranda <cmiranda@synopsys.com>
Date: Thu, 4 Oct 2018 10:17:03 +0100
Subject: [PATCH 1001/1003] [ARC] Added warning for static TLS reloc.

Signed-off-by: Alexey Brodkin <abrodkin@synopsys.com>
---
 bfd/elf32-arc.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/bfd/elf32-arc.c b/bfd/elf32-arc.c
index 11aba9f..7179860 100644
--- a/bfd/elf32-arc.c
+++ b/bfd/elf32-arc.c
@@ -2046,6 +2046,26 @@ elf_arc_check_relocs (bfd *			 abfd,
       if (is_reloc_for_GOT (howto)
 	  || is_reloc_for_TLS (howto))
 	{
+	  if (bfd_link_dll (info) && !bfd_link_pie (info)
+	      && (r_type == R_ARC_TLS_LE_32 || r_type == R_ARC_TLS_LE_S9)
+	     )
+	    {
+	      const char *name;
+	      if (h)
+	        name = h->root.root.string;
+	      else
+	        /* bfd_elf_sym_name (abfd, symtab_hdr, isym, NULL);  */
+	        name = "UNKNOWN";
+	      _bfd_error_handler
+	        /* xgettext:c-format */
+	        (_("%pB: relocation %s against `%s' can not be used"
+	           " when making a shared object; recompile with -fPIC"),
+	         abfd,
+	         arc_elf_howto (r_type)->name,
+	         name);
+	      bfd_set_error (bfd_error_bad_value);
+	      return FALSE;
+	    }
 	  if (! _bfd_elf_create_got_section (dynobj, info))
 	    return FALSE;
 
-- 
2.7.4

