From a8530198cbe6452776fe523e13c7e73b728e0fa1 Mon Sep 17 00:00:00 2001
From: Evgeniy Didin <Evgeniy.Didin@synopsys.com>
Date: Thu, 28 Mar 2019 16:47:41 +0300
Subject: [PATCH] Remove host /usr/lib and /usr/include usage

Signed-off-by: Evgeniy Didin <Evgeniy.Didin@synopsys.com>
---
 configure.ac | 2 --
 1 file changed, 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1fdeb0f..c4efedb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -240,8 +240,6 @@ fi
 dnl ==========================================================================
 dnl Common installation locations
 dnl ==========================================================================
-COMMON_INCLUDE_DIR="/usr/include /usr/local/include"
-COMMON_LIB_DIR="/usr/lib /usr/lib64 /usr/local/lib"
 case $host in
     i*86-*-linux-gnu)   COMMON_LIB_DIR="$COMMON_LIB_DIR /usr/lib/i386-linux-gnu" ;;
     x86_64-*-linux-gnu) COMMON_LIB_DIR="$COMMON_LIB_DIR /usr/lib/x86_64-linux-gnu" ;;
-- 
2.16.2

